src/Hyperledger.Aries/Features/PresentProof/DefaultProofService.cs

```diff
@@ -171,6 +171,53 @@ public virtual async Task RejectProofRequestAsync(IAgentContext agentContext, st
 
             await request.TriggerAsync(ProofTrigger.Reject);
             await RecordService.UpdateAsync(agentContext.Wallet, request);
+        }
+
+        /// <inheritdoc />
+        public async Task<bool> IsRevokedAsync(IAgentContext context, string credentialRecordId)
+        {
+            return await IsRevokedAsync(context, await RecordService.GetAsync<CredentialRecord>(context.Wallet, credentialRecordId));
+        }
+
+        /// <inheritdoc />
+        public async Task<bool> IsRevokedAsync(IAgentContext context, CredentialRecord record)
+        {
+            if (record.RevocationRegistryId == null) return false;
+            if (record.State == CredentialState.Offered || record.State == CredentialState.Requested) return false;
+            if (record.State == CredentialState.Revoked || record.State == CredentialState.Rejected) return true;
```

Comment:
Should we return IsRevoked=true when I have rejected the credential offer? technically rejected != revoked, right? Seems like we should return `false` for the `rejected` state just like offered and requested.

Reply:
Yeah, that makes sense, since it's technically not revoked, but it could be on the issuer side. Same with `Requested` state: the issuer can issue it and revoke it, while the holder doesn't have that info. I'm hoping that those that have the credential as rejected, won't be checking for it's state anyway, and if they do, they should get a negative result since they didn't accept it. A better way might be to just return and update the state, instead return `bool`.